gpflow/params/dataholders.py

```diff
@@ -40,13 +41,20 @@ class DataHolder(Parameter):
     :param fix_shape: Default value is `False` and indicates that shape
         of internal tensor does not have specific shape, in other words,
         it is None.
+    :param allow_1d: do not check shape of data input value.
```

Comment:
This is very weird argument! What is so special about 1D case, that it must go with all other fixes?